tests/integration/integration_test.go

```diff
@@ -4,24 +4,126 @@
 package integration
 
 import (
+	"math/rand"
+	"sort"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/suite"
 )
 
-func runSuiteTests(sm map[string]suite.TestingSuite) func(t *testing.T) {
-	return func(t *testing.T) {
-		for name, ts := range sm {
-			t.Run(name, func(t *testing.T) {
-				suite.Run(t, ts)
-			})
+type namedSuite struct {
+	name      string
+	testSuite suite.TestingSuite
+}
+
+// TestingGroup provides an interface to run groups of tests.
+type TestingGroup interface {
+	Name() string
+	Run(*testing.T)
+}
+
+// Prioritized is an interface for checking a type's priority.
+type Prioritized interface {
+	Priority() int
+}
+
+// Shuffler is an interface for shuffling an object's contents.
+type Shuffler interface {
+	Shuffle()
+}
+
+// TestGroup contains other tests to run.
+// Tests can be testify test suites or other child TestingGroups.
+type TestGroup struct {
+	name     string
+	prio     int
+	children []TestingGroup
+	suites   []namedSuite
+}
+
+// Run all contained tests.
+func (tg *TestGroup) Run(t *testing.T) {
+	for _, s := range tg.suites {
+		t.Run(s.name, func(t *testing.T) {
+			suite.Run(t, s.testSuite)
+		})
+	}
+	for _, child := range tg.children {
+		t.Run(child.Name(), child.Run)
+	}
+}
+
+// Name of the test group.
+func (tg *TestGroup) Name() string {
+	return tg.name
+}
+
+// Priority of the test group. Lower value is higher priority.
+func (tg *TestGroup) Priority() int {
+	return tg.prio
+}
+
+// Child returns an new child test group with the provided name.
+func (tg *TestGroup) Child(name string) *TestGroup {
+	return tg.ChildPriority(name, 1)
+}
+
+// ChildPriority returns a new child test group with the provided
+// name and priority value.
+func (tg *TestGroup) ChildPriority(name string, prio int) *TestGroup {
+	child := &TestGroup{name: name, prio: prio}
+	tg.children = append(tg.children, child)
+	sort.SliceStable(tg.children, func(i, j int) bool {
+		pi := 0
+		pj := 0
+		if p, ok := tg.children[i].(Prioritized); ok {
```

Comment:
I find it a bit confusing: could it be that 'ok == false' ? How come? and if yes, do we still get a valid sort or should we panic?

Reply:
The type of the chidlren slice holds `[]TestingGroup` rather than `[]TestGroup`. The TestingGroup interface doesn't require to also be a Prioritized interface. So you could define your own type that meets TestingGroup and isn't Prioritized. 
I had thought about making priority a property of the TestGroup by making a new type like `type childGroup struct { priority int;  g TestingGroup }` and changing the slice to `children []childGroup`.... but I think I just like interfaces so I did this approach which makes priority optional.
If you think priority should be mandatory, we can change it to the other approach I outlined here.

Reply:
OK, now I get it; in that case, no need to change it.